src/panels/config/zone/dialog-zone-detail.ts

```diff
@@ -183,7 +183,7 @@ class DialogZoneDetail extends LitElement {
 
   private _valueChanged(ev: CustomEvent) {
     this._error = undefined;
-    const value = ev.detail.value;
+    const value = { ...ev.detail.value };
```

Comment:
In the line below where `value.location` was deleted, this was actually deleting the value.location stored in ha-form's data property, causing any change in the location to glitch from oldLocation -> home.zone location -> newLocation